### DISCUSSION TITLE
Accessing resultMask in get_references()

### DISCUSSION THREAD

**Discussion by MaxWagels** [FreeOpcUa/python-opcua#1295] [Q&A]
Hello, 
I want to use resultMask to specify the output of get_references() to  the nodeId, nodeClass1 and referenceTypeId (as described [here](https://reference.opcfoundation.org/v104/Core/docs/Part4/7.25/))
How do I access the resultMask?

**Comment by AndreasHeine** [comment]
you should take a look in client.uaclient (uaclient.py)

**Reply by MaxWagels** [reply]
Thanks for your reply.
I have only found a way to specify "nodeclassmask", "direction" and "includesubtypes" but nothing to specify "reusltMask"

**Reply by AndreasHeine** [reply]
https://github.com/FreeOpcUa/python-opcua/blob/858274721766907148b527916e97446dbbf6af6d/opcua/common/node.py#L362

**Reply by AndreasHeine** [reply]
https://reference.opcfoundation.org/v104/Core/docs/Part4/5.8.2/

**Reply by MaxWagels** [reply]
I am really sorry for my stupid question, but I kind of still dont get, how to access ResultMask

**Reply by AndreasHeine** [reply]
you need to make a browse request from uaclient and pass the right BrowseDescription with your parameters


```python

        desc = ua.BrowseDescription()
        desc.BrowseDirection = ua.BrowseDirection.Both
        desc.ReferenceTypeId = _to_nodeid(ua.ObjectIds.References)
        desc.IncludeSubtypes = True
        desc.NodeClassMask = ua.NodeClass.Unspecified
        desc.ResultMask = ua.BrowseResultMask.All #<------------

        desc.NodeId = <nodeid>
        params = ua.BrowseParameters()
        params.View.Timestamp = ua.get_win_epoch()
        params.NodesToBrowse.append(desc)
        params.RequestedMaxReferencesPerNode = 0

        result = client.uaclient.browse(params) 
```

**Reply by MaxWagels** [reply]
ahhh alright, I think I got it.
Thank your very much for your help and your patience
